crepecake/src/main/java/com/google/cloud/tools/crepecake/image/Layer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+
+/**
+ * Represents a layer in an image. Implementations represent the various {@link LayerType}s.
+ *
+ * <p>An image layer consists of:
+ *
+ * <ul>
+ *   <li>Content BLOB
+ *   <li>
+ *       <ul>
+ *         <li>The compressed archive (tarball gzip) of the partial filesystem changeset.
+ *       </ul>
+ *
+ *   <li>Content Digest
+ *   <li>
+ *       <ul>
+ *         <li>The SHA-256 hash of the content BLOB.
+ *       </ul>
+ *
+ *   <li>Content Size
+ *   <li>
+ *       <ul>
+ *         <li>The size (in bytes) of the content BLOB.
+ *       </ul>
+ *
+ *   <li>Diff ID
+ *   <li>
+ *       <ul>
+ *         <li>The SHA-256 hash of the uncompressed archive (tarball) of the partial filesystem
+ *             changeset.
+ *       </ul>
+ *
+ * </ul>
+ */
+public abstract class Layer {
+
+  protected Layer() {}
+
+  /** @return the layer's {@link LayerType} */
+  public abstract LayerType getType();
+
+  /**
+   * @return the layer's content {@link BlobDescriptor}
+   * @throws LayerException if not available
+   */
+  public abstract BlobDescriptor getBlobDescriptor() throws LayerException;
+
+  /**
+   * @return the layer's diff ID
```

Comment:
this doesn't really tell me what a diff ID is though

Reply:
I have the description above in the class javadoc - should I duplicate that to here as well?

Reply:
No, that's good. Thanks.